TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/EWCWalesImportJob.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            StartDate = DateTime.Now,
+            TypeId = (int)dfeta_IntegrationInterface.GTCWalesImport,
+            FileName = fileName
+        };
+        var integrationId = await crmQueryDispatcher.ExecuteQuery(integrationJob);
+        var importType = GetImporFileType(fileName);
+        var (totalCount, successCount, duplicateCount, failureCount, failureMessage) = (0, 0, 0, 0, "");
+
+        if (importType == EWCWalesImportFileType.Induction)
+        {
+            var results = await inductionImporter.Import(reader, integrationId, fileName);
+            totalCount = results.TotalCount;
+            successCount = results.SuccessCount;
+            duplicateCount = results.DuplicateCount;
+            failureCount = results.FailureCount;
+            failureMessage = results.FailureMessage;
+        }
+        else if (importType == EWCWalesImportFileType.Qualification)
+        {
+            var results = await qtsImporter.Import(reader, integrationId, fileName);
+            totalCount = results.TotalCount;
+            successCount = results.SuccessCount;
+            duplicateCount = results.DuplicateCount;
+            failureCount = results.FailureCount;
+            failureMessage = results.FailureMessage;
+        }
+        else
+        {
+            totalCount = 0;
+            successCount = 0;
+            duplicateCount = 0;
+            failureCount = 0;
+            failureMessage = "Import filename must begin with IND or QTS";
+        }
+
+        var updateIntegrationTransactionQuery = new UpdateIntegrationTransactionQuery()
+        {
+            IntegrationTransactionId = integrationId,
+            EndDate = DateTime.Now,
+            TotalCount = totalCount,
+            SuccessCount = successCount,
+            DuplicateCount = duplicateCount,
+            FailureCount = failureCount,
+            FailureMessage = failureMessage
+        };
+        await crmQueryDispatcher.ExecuteQuery(updateIntegrationTransactionQuery);
+        return (totalCount, successCount, duplicateCount, failureCount, failureMessage, integrationId);
+    }
+
+    public StreamReader GetStreamReader(byte[] filecontents)
+    {
+        var stream = new MemoryStream(filecontents);
+        var reader = new StreamReader(stream);
+        return reader;
+    }
+
+    public async Task Execute(CancellationToken cancellationToken)
+    {
+        foreach (var file in GetImportFiles())
+        {
+            using (var downloadStream = await GetDownloadStream(file))
+            using (var reader = GetStreamReader(downloadStream.ToArray()))
+            {
+                await Import(file, reader);
+            }
+        }
+    }
+
+    public EWCWalesImportFileType GetImporFileType(string fileName) => fileName.ToLower() switch
+    {
+        var filename when filename.Contains("Ind", StringComparison.OrdinalIgnoreCase) => EWCWalesImportFileType.Induction,
```

Comment:
It's a little confusing to have both `fileName` and `filename`
```suggestion
        var f when f.Contains("Ind", StringComparison.OrdinalIgnoreCase) => EWCWalesImportFileType.Induction,
```